### DISCUSSION TITLE
Geometry node cartoon utilities

### DISCUSSION THREAD

**Discussion by chabi-fin** [BradyAJohnston/MolecularNodes#377] [General]
Awesome tool! Has anyone been able to get the cartoon utilities working within the geometry node? It requires a profile curve, any clue what this is?

**Comment by BradyAJohnston** [comment]
It's the node group that powers the Cartoon style underneath, just with more options exposed.  You don't _require_ the profile curve, but you do require `sec_struct` attribute that is properly calculated on import (usually best when downloading from the PDB).

**Reply by chabi-fin** [reply]
Hey thanks a lot for the tip, it helped and I appreciate it; My problem turned out that I was connecting the cartoon in series with the default cartoon settings, when I really just needed to replace the node!
I'm still having an issue with the appearance of the cartoon, where strange kinks appear even though the `sec_struct` attribute is correctly assigned. I checked inside the vertex attributes and can see the structure assignment is correct, but the structure still appears terrible. Messing around with the resolution and other settings didn't help either. 
As you can see, this example segment makes a nice turn in Pymol, but looks very strange in blender. Any ideas on this?
![helix-blender](https://github.com/BradyAJohnston/MolecularNodes/assets/69977647/772a1ff9-2305-46a9-8718-21174f00ccba)
![helix-pymol](https://github.com/BradyAJohnston/MolecularNodes/assets/69977647/8173f70f-8025-4110-8ccf-1259eccc34bc)

**Reply by BradyAJohnston** [reply]
Are you able to share the PDB code this happens to? I have seen it happen on a couple of structures, and is a problem with the underlying node tree that makes the cartoon. I have been meaning to fix it, but I need some more concrete examples to test against.

**Reply by chabi-fin** [reply]
The original PDB code would be 3PEE, and if I use just that structure the cartoon looks fine. Mine is actually a conformational morph using Chimera, with the same HELIX and SHEET lines as in the 3PEE PDB file. So I think the geometry nodes are having a problem when it needs to map the designation to multiple conformations? I attached the multi-frame pdb file if you would like to see how it behaves.
[morph_confs_apo.zip](https://github.com/BradyAJohnston/MolecularNodes/files/13686514/morph_confs_apo.zip)

**Reply by BradyAJohnston** [reply]
Thanks for the files. I have opened an issue #382 to track this and will see how I go with improving the cartoon styles.
